### PR TITLE
Fix lint apiversion v3

### PR DESCRIPTION
**Pull request by mattfarina**
Porting #5729 to Helm v3

### CODE REVIEW

File: pkg/lint/rules/chartfile.go [hunk 1]
```diff
@@ -85,6 +87,18 @@ func validateChartNameDirMatch(chartDir string, cf *chart.Metadata) error {
 	return nil
 }
 
+func validateChartAPIVersion(cf *chart.Metadata) error {
+	if cf.APIVersion == "" {
```
Comment (bacongobbler) [inline pkg/lint/rules/chartfile.go]: Could we somehow tie into [`cf.Validate()`](https://github.com/helm/helm/blob/55ae0c2efb48751cefe51cd66567608c0f7a650e/pkg/chart/metadata.go#L71) instead? That way we could keep the chart metadata validation DRY and in one package. 

Perhaps we could change the error output in `.Validate` to return typed errors like `ErrNoAPIVersionFound` or `ErrInvalidAPIVersion`. What do you think about that?

File: pkg/lint/lint_test.go [hunk 2]
```diff
@@ -62,9 +62,13 @@ func TestBadChart(t *testing.T) {
 			if strings.Contains(msg.Err.Error(), "directory name (badchartfile) and chart name () must be the same") {
 				e3 = true
 			}
+
+			if strings.Contains(msg.Err.Error(), "apiVersion is required") {
```
Comment (bacongobbler) [inline pkg/lint/lint_test.go]: thought: the comment below about typed error values from `.Validate` could be a nice refactor here.

File: pkg/lint/rules/chartfile.go (hunk 1 above)
Comment (mattfarina) [reply pkg/lint/rules/chartfile.go]: I like the idea of typing into the `Validate` function in the chart package. From the looks of things, the current function is simple compared to what linting is doing on the `Chart.yaml` file. Other parts of linting have not been moved over to `Validate` either.

Should we do that migration in a separate PR? One that handles the change wholesale.

Comment (bacongobbler) [reply pkg/lint/rules/chartfile.go]: Yeah, we can do that in a separate PR. 👍

**Review by bacongobbler — APPROVED** [review]

File: pkg/lint/lint_test.go (hunk 2 above)
Comment (mattfarina) [reply pkg/lint/lint_test.go]: I added the guidance as you suggested.

**Pull request merged**